scripts/release-cdn.mts

```diff
@@ -0,0 +1,150 @@
+/**
+ *
+ * Copyright 2020-2025 Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import crypto from 'crypto'
+import * as fs from 'fs/promises'
+import * as path from 'path'
+import dotenv from 'dotenv'
+
+import {
+	generateCDNLinks,
+	generateScriptSnippet,
+	getPositionalArguments,
+	invalidateCloudFrontFiles,
+	patchGithubReleaseBody,
+	uploadToS3,
+	getReleaseForTag,
+	getAllVersions,
+	getMimeType,
+} from './utils/index.mjs'
+
+const CDN_LISTED_FILES = ['splunk-otel-web.js', 'splunk-otel-web-session-recorder.js']
+const ARTIFACTS_DIR = './artifacts'
+const OWNER = process.env.GITHUB_OWNER ?? 'signalfx'
+const REPO = process.env.GITHUB_REPO ?? 'splunk-otel-js-web'
+
+const isDryRun = process.argv.some((arg) => arg === '--dry-run')
+if (isDryRun) {
+	console.log('---- DRY RUN ----')
+}
+
+dotenv.config()
+
+const [targetVersion] = getPositionalArguments()
+
+if (!targetVersion) {
+	throw new Error('You must provide a version to release.')
+}
+
+if (!process.env.CDN_DISTRIBUTION_ID) {
+	throw new Error('You are missing an environment variable CDN_DISTRIBUTION_ID.')
+}
+
+const { CDN_DISTRIBUTION_ID } = process.env
+
+if (!process.env.CDN_BUCKET_NAME) {
+	throw new Error('You are missing an environment variable CDN_BUCKET_NAME.')
+}
+
+const { CDN_BUCKET_NAME } = process.env
+
+if (!process.env.GITHUB_TOKEN && !isDryRun) {
+	throw new Error('You are missing an environment variable GITHUB_TOKEN.')
+}
+
+console.log('I will now process the files:')
+const cdnLinksByVersion: Record<string, string[]> = {}
+
+// Avoid accidental assets upload
+const allowedExtensions = ['.tgz', '.js', '.js.map', '.txt']
+const assets = await fs.readdir(ARTIFACTS_DIR)
```

Comment:
We take the artifacts from the build step instead of downloading them from GitHub.